src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -3613,16 +3613,22 @@ ReferenceTableOidList()
 	while (HeapTupleIsValid(heapTuple))
 	{
 		bool isNull = false;
-		Datum relationIdDatum = heap_getattr(heapTuple,
-											 Anum_pg_dist_partition_logicalrelid,
-											 tupleDescriptor, &isNull);
-		Oid relationId = DatumGetObjectId(relationIdDatum);
 		char partitionMethod = heap_getattr(heapTuple,
 											Anum_pg_dist_partition_partmethod,
 											tupleDescriptor, &isNull);
+		char replicationModel = heap_getattr(heapTuple,
+											 Anum_pg_dist_partition_repmodel,
+											 tupleDescriptor, &isNull);
 
-		if (partitionMethod == DISTRIBUTE_BY_NONE)
+		if (partitionMethod == DISTRIBUTE_BY_NONE &&
```

Comment:
maybe create a utility `isReferenceTable(partitionMethod,replicationModel)` so that we remove the mapping and directly include it in our code

Reply:
Maybe in this pr or in a seperate pr, we can refactor this function to call `IsReferenceTable` function for the oids in `pg_dist_partition` ?

Does it sound good ?

Reply:
sure